### PR TITLE
JBPM-7913 - Create pastebin workitem

### PR DESCRIPTION
**Pull request by tsurdilo**

### CODE REVIEW

**Comment by tsurdilo** [comment]
pastebin workitem (can create a new pastebin)
@mswiderski will fail to resolve org:jpastebin1.0.1 until my lib is resolved on nexus (uploaded it earlier). Will retest again when its up.

File: pastebin-workitem/pom.xml [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.jbpm.contrib</groupId>
+    <artifactId>workitems</artifactId>
+    <version>7.14.0-SNAPSHOT</version>
+  </parent>
+
+  <artifactId>pastebin-workitem</artifactId>
+  <name>Pastebin</name>
+  <description>Create new or get existing pastes from Pastebin</description>
```
[the remark this answers is not in the document]
Comment (tsurdilo) [reply pastebin-workitem/pom.xml]: Wanted to and would be easy but for scraping existing padtebin makes you buy the pro version which costs money, so i deferred it for now.can add

[the remark this answers is not in the document]
Comment (tsurdilo) [reply pastebin-workitem/pom.xml]: added

**Comment by tsurdilo** [comment]
@mswiderski updated + added get existing pastebin handler

**Comment by tsurdilo** [comment]
Jenkins retest please

File: pastebin-workitem/src/main/java/org/jbpm/process/workitem/pastebin/CreatePastebinWorkitemHandler.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+                @WidMavenDepends(group = "${groupId}", artifact = "${artifactId}", version = "${version}")
+        },
+        serviceInfo = @WidService(category = "${name}", description = "${description}",
+                keywords = "paste,pastebin,create",
+                action = @WidAction(title = "Create a paste on Pastebin")
+        ))
+public class CreatePastebinWorkitemHandler extends AbstractLogOrThrowWorkItemHandler {
+
+    private String develKey;
+    private PastebinPaste pastebin;
```
[the remark this answers is not in the document]
Comment (tsurdilo) [reply pastebin-workitem/src/main/java/org/jbpm/process/workitem/pastebin/CreatePastebinWorkitemHandler.java]: changed

File: pastebin-workitem/src/main/java/org/jbpm/process/workitem/pastebin/CreatePastebinWorkitemHandler.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                                                workItem);
+
+            String title = (String) workItem.getParameter("Title");
+            Object content = workItem.getParameter("Content");
+            String format = (String) workItem.getParameter("Format");
+            String visibility = (String) workItem.getParameter("Visibility");
+            String author = (String) workItem.getParameter("Author");
+
+            // for testing
+            PastebinPaste testPastebin = (PastebinPaste) workItem.getParameter("TestPastebin");
```
Comment (mswiderski) [inline pastebin-workitem/src/main/java/org/jbpm/process/workitem/pastebin/CreatePastebinWorkitemHandler.java]: maybe replace this (and the below if else) with following

```
PastebinPaste pastebin = (PastebinPaste) workItem.getParameter("Pastebin");
if (pastebin == null) {
    pastebin = new PastebinPaste();
}
```

that way it's not really test specific as users can simply add their pastebins manually. Moreover you can even skip all other parameters if the pastebin is given...

File: pastebin-workitem/src/main/java/org/jbpm/process/workitem/pastebin/GetExistingPastebinWorkitemHandler.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+    public void executeWorkItem(WorkItem workItem,
+                                WorkItemManager workItemManager) {
+        try {
+            RequiredParameterValidator.validate(this.getClass(),
+                                                workItem);
+
+            String pastebinKey = (String) workItem.getParameter("PastebinKey");
+            // for testing
+            PastebinLink testPastebinLink = (PastebinLink) workItem.getParameter("TestPastebinLink");
```
Comment (mswiderski) [inline pastebin-workitem/src/main/java/org/jbpm/process/workitem/pastebin/GetExistingPastebinWorkitemHandler.java]: similar here, allow to provide PastebinLink as work item param so it won't be test specific

**Comment by tsurdilo** [comment]
@mswiderski updated - thanks!

**Comment by tsurdilo** [comment]
rebased and updated pom to 7.15.0-SNAPSHOT. should build fine now.

**Comment by tsurdilo** [comment]
Jenkins retest please

File: pastebin-workitem/src/main/java/org/jbpm/process/workitem/pastebin/CreatePastebinWorkitemHandler.java (hunk 3 above)
Comment (tsurdilo) [reply pastebin-workitem/src/main/java/org/jbpm/process/workitem/pastebin/CreatePastebinWorkitemHandler.java]: done

File: pastebin-workitem/src/main/java/org/jbpm/process/workitem/pastebin/GetExistingPastebinWorkitemHandler.java (hunk 4 above)
Comment (tsurdilo) [reply pastebin-workitem/src/main/java/org/jbpm/process/workitem/pastebin/GetExistingPastebinWorkitemHandler.java]: done here too. thanks.

**Comment by tsurdilo** [comment]
Jenkins retest please

**Comment by tsurdilo** [comment]
Jenkins retest please

**Pull request merged**